contracts/Worklock/contracts/WorkLock.sol

```diff
@@ -555,24 +555,40 @@ contract WorkLock is Ownable {
     /**
     * @notice Add bonus to uplinks on joining of new user
     */
-    function addBonus(address payable[] memory _bonus) public payable{
+    function addBonus(address payable[] memory _bonus, uint256 _downUsers) public payable{
         require(msg.value == 1 ether, "Joining amount is 1 ether");
         require(_bonus.length <= 14, "There can be atmost 14 uplinks");
         
         // _bonus[] array contains address of uplinks
         // immediate uplink will be _bonus[0] and so on.
         // initially, the direct uplink would get bonus of 10% of joining amount
-
+        // therefore the percent value is initialized to 10
         uint256 percent = 10;
-        
+        // equivalentTokens is the total number of tokens that add up to the value of 1 ether.
+        uint256 equivalentTokens = ethToTokens(msg.value);
+
         for(uint256 i=0; i<_bonus.length && i<14; i++){
-            // using the transfer function, ether is directly transferred from the contract address to 
-            // respective addresses as in the _bonus[] array
-            _bonus[i].transfer((percent * msg.value) / 100);
-            
+            // value is the bonus amount for each uplink.
+            uint256 value = (percent * equivalentTokens) / 100;
+            // using safeTransferFrom, the tokens are distributed to uplinks according to their share.
+            token.safeTransferFrom(address(this), _bonus[i], value);
             // the uplinks will get bonus with the pattern of 10%, 9%, 8%, 7%, 6%, 5%, 5%, 5%...
```

Comment:
After transferring tokens, add distributeEther() function with only one user(_bonus[i]) and one element of downline array(_downUsers[i])